### ISSUE TITLE
Feature Request: add ability to change live stream visibility

### ISSUE THREAD

**Issue by mark-epstein** [bitfocus/companion-module-youtube-live#100]
We schedule a weekly YouTube stream several days to a week in advance, but we want it to remain invisible until about 24 hours before we go live. 
I'd like to automate the last step of changing our stream from "unlisted" to "public" for two reasons:
1. eliminates some potential user error, e.g. forgetting to update the visibility setting
2. someone doesn't have to do extra work on a Saturday to prepare for a Sunday stream

~My intuition tells me that it's not possible since it seems that "unlisted" videos don't show up anywhere in the module, which makes me think they are invisible to the API. But maybe the developers just didn't see any reason (until now!) to query unlisted scheduled videos.~ deleted for its incorrectness

**Comment by cyprient** [comment]
Hello @mark-epstein,

I just checked on my configuration, and I can see my _unlisted_ videos. So you should see yours in the variable list of the module.

**Comment by mark-epstein** [comment]
> I can see my _unlisted_ videos. So you should see yours in the variable list of the module
You are correct, I do. I was relying on apparently faulty memory when I suggested that YouTube might hide the unlisted streams from the API.

**Comment by mark-epstein** [comment]
As an alternative or additional suggestion, maybe add a variable such as $(YT:unfinished_0_visibility)
So I can make Companion take an action at a given time if the scheduled stream isn't public.

**Comment by jswalden** [comment]
This is fixed in #123, which will be in https://github.com/bitfocus/companion-module-youtube-live/releases/tag/v2.5.0 that's submitted for acceptance in Companion 4.0.

To be precise, this adds only the action to change visibility as the summary requests -- not any variables exposing visibility.  I don't see anything in API docs or implementation that makes it problematic to run the action to "change" visibility when in fact the already-used visibility is specified.  People can feel free to file a new issue for variables if they want, then wait for a fix to be proposed.  (I'm a vague somewhat-interloper in this module who is willing to review stuff, and perhaps willing to write patches for my own issues -- but it's very unlikely I would go out of my way to write a patch for others' issues.)
